### DISCUSSION TITLE
Can't upgrade due to salt version not upgrading

### DISCUSSION THREAD

**Discussion by alitheg** [Security-Onion-Solutions/securityonion#4116] [Unsupported Versions]
I think I've resolved this but wanted to raise it as a potential bug. It looks like the current version replaces the existing individual yum repos with a securityonion repo for each package, ensuring the old ones are gone: https://github.com/Security-Onion-Solutions/securityonion/pull/3855/files

However, I can't get that to apply via soup because it tries to check that salt 3003 has been installed, which it can't because the saltstack.repo file I have is pinned at 3002. Therefore, the upgrade never proceeds. I manually replaced the saltstack.repo file to use 3003 from the official sources, and so far things look better.

**Comment by Sazerak** [comment]
I ran into the same issue, and I'm not sure if my solution was the same as yours, but I simply modified the file /etc/yum.repos.d/saltstack.repo to link to the 3003 version instead of 3002.2 (both the URL and GPG key lines). Still updating, but so far so good.

**Comment by alitheg** [comment]
Yup, that's exactly what I did @Sazerak - then the release proceeded with the updated version and it later removed that file as expected.

**Comment by Sazerak** [comment]
@alitheg Did the rest of your update go smoothly? Mine has been stuck on "Executing command systemd-run in directory '/root' " for about 10 minutes now.

**Comment by alitheg** [comment]
Yeah - I didn't watch it in detail so I'm not sure how long each step took, but it did take some time. Worked eventually!

**Comment by alitheg** [comment]
@TOoSmOotH Thanks for moving to a discussion - is there a bug in here around upgrading when the salt version has been pinned by an older version in the repo file, or is something wrong with our deployments that's caused it?

**Reply by TOoSmOotH** [reply]
the repo.client state should run before it tries to install the new salt. What happen when you so a sudo salt-call state.apply repo.client
